numba/npyufunc/parfor.py

```diff
@@ -513,8 +513,23 @@ def compute_def_once_block(block, def_once, def_more, getattr_taken):
             # Calling a method on an object is like a def of that object.
             base_obj = getattr_taken[rhs.func.name]
             add_to_def_once_sets(base_obj, def_once, def_more)
-
-def compute_def_once_internal(loop_body, def_once, def_more, getattr_taken):
+        if isinstance(rhs, ir.Expr) and rhs.op == 'call':
+            # If a mutable object is passed to a function, then it may be changed and
+            # therefore can't be hoisted.
+            # For each argument to the function...
+            for argvar in rhs.args:
+                # Get the argument's type.
+                if isinstance(argvar, ir.Var):
+                    argvar = argvar.name
+                avtype = typemap[argvar]
+                # If that type doesn't have a mutable attribute or it does and it's set to
+                # not mutable then this usage is safe for hoisting.
+                if hasattr(avtype, 'mutable') and avtype.mutable:
```

Comment:
Is this the same as:
```
if getattr(avtype, 'mutable', False):
```
?

Reply:
Thanks.  I'll remember the idiom one of these days.